src/display/annotation_layer.js

```diff
@@ -441,31 +440,92 @@ class AnnotationElement {
    * Create quadrilaterals from the annotation's quadpoints.
    *
    * @private
-   * @param {boolean} ignoreBorder
    * @memberof AnnotationElement
-   * @returns {Array<HTMLElement>} An array of section elements.
    */
-  _createQuadrilaterals(ignoreBorder = false) {
-    if (!this.data.quadPoints) {
-      return null;
-    }
-
-    const quadrilaterals = [];
-    const savedRect = this.data.rect;
-    let firstQuadRect = null;
-    for (const quadPoint of this.data.quadPoints) {
-      this.data.rect = [
-        quadPoint[2].x,
-        quadPoint[2].y,
-        quadPoint[1].x,
-        quadPoint[1].y,
+  _createQuadrilaterals() {
+    if (!this.container) {
+      return;
+    }
+    const { quadPoints } = this.data;
+    if (!quadPoints) {
+      return;
+    }
+
+    const [rectBlX, rectBlY, rectTrX, rectTrY] = this.data.rect;
+
+    if (quadPoints.length === 1) {
+      const [, { x: trX, y: trY }, { x: blX, y: blY }] = quadPoints[0];
+      if (
+        rectTrX === trX &&
+        rectTrY === trY &&
+        rectBlX === blX &&
+        rectBlY === blY
+      ) {
+        // The quadpoints cover the whole annotation rectangle, so no need to
+        // create a quadrilateral.
+        return;
+      }
+    }
+
+    const { style } = this.container;
+    const hasBorder = style.borderWidth && style.borderWidth !== "0px";
```

Comment:
Could we not do this without having to (unconditionally) query the DOM-element?
The simplest solution might be to add a property such as `this._hasBorder = true;` e.g. [in this block](https://github.com/mozilla/pdf.js/blob/dca4bc0f24cd755f5e22b6992dd99fcf8066a6af/src/display/annotation_layer.js#L281-L286) and then check that in this method instead.